internal/configs/static.go

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package configs
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/opentofu/opentofu/internal/lang"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type StaticIdentifier struct {
+	Module string
+	Type   string
+	Name   string
+}
+
+func (ref StaticIdentifier) String() string {
+	val := ref.Name
+	if len(ref.Type) != 0 {
+		val = ref.Type + "." + val
+	}
+	if len(ref.Module) != 0 {
+		val = ref.Module + "." + val
+	}
+	return val
+}
+
+type StaticReference struct {
```

Comment:
Please add comments (ideally with short examples embedded) of what each of these structs are.

Reply:
Removed